pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -604,6 +625,24 @@ private void doAuthentication(PGStream pgStream, String host, String user, Prope
                 sspiClient.continueSSPI(l_msgLen - 8);
                 break;
 
+              case AUTH_REQ_SASL:
+                if (LOGGER.isLoggable(Level.FINEST)) {
+                  LOGGER.log(Level.FINEST, " <=BE AuthenticationSASL");
```

Comment:
There's no need to verify `isLoggable` when arguments to `LOGGER.log` are trivial. In this case, message is just passed as is, and no arrays created, so isLoggable should be just dropped.